docs/data/tree-view/rich-tree-view/lazy-loading/lazy-loading.md

```diff
@@ -0,0 +1,36 @@
+---
+productId: x-tree-view
+title: Rich Tree View - Lazy Loading Children
+components: RichTreeView, TreeItem
+packageName: '@mui/x-tree-view'
+githubLabel: 'component: tree view'
+waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
+---
+
+# Rich Tree View - Lazy Loading Children
```

Comment:
```suggestion
# Rich Tree View - Lazy loading children [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
```

Reply:
Yep I agree, it should be part of the pro plan

Reply:
https://github.com/mui/mui-x/issues/15373

Reply:
So why have data grid data source MIT but tree view data source Pro?

- In the case of the data grid data: data source allows to maximize the open source adoption **and** even if you have the feature, you can't do much with it, you still need the Pro plan for their features. So MIT.
- In the case of the Tree View: I would expect developers to be able to implement lazy loading MIT using React Query (https://github.com/TanStack/query/discussions/413), I assumed that this page is about doing more than this. But looking closer, it's not so clear. So maybe the right move is:

1. Make the page MIT, since we likely want to have an example using TanStack Query https://npm-stat.com/charts.html?package=react-dom&package=%40tanstack%2Freact-query&from=2023-11-12&to=2024-11-12? 
2. Add that example
3. Rely on the fact that once people integrate lazy loading, they are more likely to hit a point where they need virtualization, and we need Lazy loading to be competitive in the MIT space, e.g. https://ant.design/components/tree#tree-demo-dynamic. This is a better 📈 ramp.
4. We see if there is space for a more complex data fetching logic. Maybe we don't need a data source, if we do (Pro). Data source could truly shine for cases where:
 
 - the simple lazy loading API isn't enough
 - you are using a data grid data source, you are used to it, and you want the same abstraction
 - you are going to use virtualization, and you need to be smart about loading
 - SWR or React Query have too many limitations (we need to be careful to not rebuild those).